uitools/toolkitcpp/import/Esri/ArcGISRuntime/Toolkit/ArcGISUserCredentialsView.qml

```diff
@@ -0,0 +1,143 @@
+
+import Esri.ArcGISRuntime.Toolkit.Controller
+
+import QtQuick
+import QtQuick.Controls
+import QtQuick.Layouts
+
+Dialog {
+    id: userCredentialsView
+
+    property ArcGISAuthenticationController controller: null
+
+    title: qsTr("Authentication required")
+
+    closePolicy: Popup.NoAutoClose
+
+    header: Label {
+        text: `<h1>${title}</h1>`
+        horizontalAlignment: Text.AlignHCenter
+        padding: 5
+    }
+
+    footer: DialogButtonBox {
+        Button {
+            text: qsTr("Skip")
+            DialogButtonBox.buttonRole: DialogButtonBox.RejectRole
+        }
+
+        Button {
+            text: qsTr("Continue")
+            enabled: hasValidInput()
+            DialogButtonBox.buttonRole: DialogButtonBox.AcceptRole
+        }
+
+        onAccepted: {
+            acceptWithCurrentUsernameAndPassword();
+        }
+
+        onRejected: {
+            controller.cancel();
+        }
+    }
+
+    ColumnLayout {
+        anchors.centerIn: parent
+        Label {
+            text: qsTr("Invalid username or password.")
+            horizontalAlignment: Qt.AlignHCenter
+            visible: controller.currentChallengeFailureCount > 0
+            color: "red"
+            background:  Rectangle {
+                color: "#FFCCCC"
+                radius: 5
+            }
+            Layout.fillWidth: true
+            Layout.margins: 5
+        }
+
+        Label {
+            text: controller.currentAuthenticatingHost
+            horizontalAlignment: Qt.AlignHCenter
+            font.weight: Font.Bold
+            Layout.fillWidth: true
+            Layout.margins: 5
+        }
+
+        Label {
+            text: "You need to sign in to access this resource."
+            horizontalAlignment: Qt.AlignHCenter
+            wrapMode: Text.Wrap
+            Layout.fillWidth: true
+            Layout.margins: 5
+        }
+
+        TextField {
+            id: usernameTextField
+            placeholderText: qsTr("username")
+            onAccepted: acceptWithCurrentUsernameAndPassword()
+            Layout.fillWidth: true
+            Layout.margins: 5
+        }
+
+        TextField {
+            id: passwordTextField
+            placeholderText: qsTr("password")
+            onAccepted: acceptWithCurrentUsernameAndPassword()
+            echoMode: TextInput.Password
+            Layout.fillWidth: true
+            Layout.margins: 5
+        }
+    }
+
+    function hasValidInput() {
+        return usernameTextField.text !== "" && passwordTextField.text !== "";
+    }
+
+    /*!
+      \internal
+      \brief pre-fill the username and password fields
+     */
+    function prefillUsernameAndPassword(username, password) {
+        usernameTextField.text = username;
+        passwordTextField.text = password;
+    }
+
+    /*!
+      \internal
+      \brief Fill and accept the username and password
+     */
+    function applyUsernameAndPassword(username, password) {
```

Comment:
You are right, this is wrong. The accept logic I ported from the existing UI control doesn't work the same with the new workflow, so I had to make some small tweaks. You are right that it isn't needed here, but we need to call accept in at least one exit point, otherwise the dialogs keep piling up on top of each other and aren't dismissed.

I will fix this and rerun all tests. Good catch.